spec/spec_helper.cr

```diff
@@ -114,22 +114,23 @@ def assert_no_errors(*args)
   semantic(*args)
 end
 
-def warnings_result(code)
-  output_filename = Crystal.temp_executable("crystal-spec-output")
-
+def warnings_result(code, *, file = __FILE__)
   compiler = create_spec_compiler
   compiler.warnings = Warnings::All
   compiler.error_on_warnings = false
   compiler.prelude = "empty" # avoid issues in the current std lib
   compiler.color = false
   apply_program_flags(compiler.flags)
-  result = compiler.compile Compiler::Source.new("code.cr", code), output_filename
 
-  result.program.warning_failures
+  with_temp_executable("crystal-spec-output", file: file) do |output_filename|
+    result = compiler.compile Compiler::Source.new("code.cr", code), output_filename
+
+    return result.program.warning_failures
```

Comment:
I don't like this suggestion, too much is implicit there. The reader needs to first realize that this line has anything to do with the return value. Additionally, we'd actually be returning this from the *block* and then we (and any future reader) would need to double-check that the block actually propagates that value.

Generally please keep in mind that because something can be omitted, doesn't mean it should be.